### PR TITLE
allow compositional fields to be advected with the melt velocity

### PR DESCRIPTION
**Pull request by jdannberg**

### CODE REVIEW

File: source/simulator/parameters.cc [hunk 1]
```diff
@@ -884,7 +884,7 @@ namespace aspect
                          Patterns::List(Patterns::Anything()),
                          "A user-defined name for each of the compositional fields requested.");
       prm.declare_entry ("Compositional field methods", "",
-                         Patterns::List (Patterns::Selection("field|particles|static")),
+                         Patterns::List (Patterns::Selection("field|particles|static|melt field")),
```
Comment (tjhei) [inline source/simulator/parameters.cc]: is there a reason the enum is called differently?

File: source/simulator/parameters.cc [hunk 2]
```diff
@@ -913,6 +913,16 @@ namespace aspect
                          "this way, then it does not evolve at all. Its values are "
                          "simply set to the initial conditions, and will then "
                          "never change."
+                         "\n"
+                         "\\item ``melt field'': If a compositional field is marked with this "
```
Comment (tjhei) [inline source/simulator/parameters.cc]: it also includes diffusion, right?

**Comment by tjhei** [comment]
Looks good. Can you fix the test output?

File: source/simulator/parameters.cc (hunk 1 above)
Comment (jdannberg) [reply source/simulator/parameters.cc]: What do you mean? The reason why the enum has a different name than the input parameter? I have no idea, but I think AdvectionFieldMethod makes a lot more sense as a name, and we can not change the name of the input parameter without breaking backwards compatibility.

File: source/simulator/parameters.cc (hunk 2 above)
Comment (jdannberg) [reply source/simulator/parameters.cc]: No, I don't think so. For compositional fields the diffusivity is always zero.

File: source/simulator/parameters.cc (hunk 1 above)
Comment (tjhei) [reply source/simulator/parameters.cc]: no, I mean why is the parameter "melt field" while the enum is "fem melt field"

File: source/simulator/parameters.cc (hunk 2 above)
Comment (tjhei) [reply source/simulator/parameters.cc]: oh. :-)

File: source/simulator/parameters.cc (hunk 1 above)
Comment (jdannberg) [reply source/simulator/parameters.cc]: Oh, I just copied the existing structure. For the normal compositional field, the parameter is "field", and the enum is "fem_field".

**Comment by jdannberg** [comment]
I think I addressed all comments, and I updated the test output, but I have no idea what is going on with the jenkins tester...

**Comment by gassmoeller** [comment]
The jenkins tester is sick at the moment, lets ignore it for now. The rest looks good.

**Comment by jdannberg** [comment]
It's sick? I hope it gets well soon :)

**Pull request merged**